pkg/agent/watch.go

```diff
@@ -215,28 +201,22 @@ func (e vmEvent) Format(state fmt.State, verb rune) {
 	}
 }
 
-type vmResourceTuple struct {
-	namespace  string
-	name       string
-	endpointID string
-	resource   vmResource
-	valueType  vmResourceValueType
+type kv[K any, V any] struct {
+	key   K
+	value V
 }
 
-func metricsForVM(vm *vmapi.VirtualMachine, nodeName string) map[vmResourceTuple]float64 {
+func setMetricsFromVM(vm *vmapi.VirtualMachine, nodeName string, perVMMetrics *PerVMMetrics) {
```

Comment:
I think the name `From` here is redundant and could be confusing. I suggest a shorter name such as `setVMMetrics`.